sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogSuite.scala

```diff
@@ -278,8 +278,18 @@ abstract class ExternalCatalogSuite extends SparkFunSuite {
   }
 
   test("get tables by name") {
-    assert(newBasicCatalog().getTablesByName("db2", Seq("tbl1", "tbl2"))
-      .map(_.identifier.table) == Seq("tbl1", "tbl2"))
+    val catalog = newBasicCatalog()
+    val tables = catalog.getTablesByName("db2", Seq("tbl1", "tbl2"))
+    assert(tables.map(_.identifier.table) == Seq("tbl1", "tbl2"))
+    assert(tables.forall(_.identifier.catalog.isDefined))
+
+    // After renaming a table, the identifier should still be qualified with catalog.
+    catalog.renameTable("db2", "tbl1", "tblone")
+    // TODO: HMS seems to have a bug and the order of returned tables do not match the order of
+    //       table names. You can reproduce the bug by passing `Seq("tblone", "tbl2")` here.
```

Comment:
cc @somani @yaooqinn @wangyum

Reply:
Yes. HMS can't guarantee of ordering of the returned tables:
https://github.com/apache/hive/blob/rel/release-2.3.9/metastore/src/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java#L383

Reply:
The table names in the request work as filters only, and they don't determine the result set order. So, it's not a bug to me.

Reply:
Ah I see, so the caller side should do the name matching. Let me remove this TODO